types/coin.go

```diff
@@ -576,8 +593,9 @@ func (coins Coins) Sort() Coins {
 // Parsing
 
 var (
-	// Denominations can be 3 ~ 64 characters long.
-	reDnmString = `[a-z][a-z0-9/]{2,63}`
+	// Denominations can be 3 ~ 128 characters long and support letters, followed by either
+	// a letter, a number or a separator ('/').
```

Comment:
Note: There's the potential for confusing UI here.

Suppose I create a denom `atom10`, the following `Coins` string representation is ambigious to an end user

5atom10,300btc

Reply:
unfortunately, there's no workaround for this. I added a test case and documented the expected format on the `ParseCoin(s)` func